macro/src/generate.rs

```diff
@@ -967,6 +974,72 @@ fn generate_to_primitive_traits(item: &BoundedInteger, tokens: &mut TokenStream)
     }
 }
 
+fn generate_index_traits(item: &BoundedInteger, tokens: &mut TokenStream) {
+    let ident = &item.ident;
+
+    tokens.extend(quote! {
+        impl<T> core::ops::Index<#ident> for [T] {
```

Comment:
It’s best to replace the usages of `core::` here with `::core::`, because that avoids conflicts if the user happens to have a module named `core` also in scope (`::core` will unambiguously refer to the _crate_ `core` rather than anything else).